driver-core/src/main/com/mongodb/internal/operation/ChangeStreamBatchCursor.java

```diff
@@ -108,8 +110,8 @@ public Boolean apply(final AggregateResponseBatchCursor<RawBsonDocument> queryBa
 
     @Override
     public void close() {
-        if (!closed) {
-            closed = true;
+        if (!closed.getAndSet(true)) {
```

Comment:
Made idempotent as per the specification of `BatchCursor.close`. Now the `close` method in all our cursors is idempotent.